core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -402,6 +403,7 @@ class TransactionCoordinator(brokerId: Int,
 
               preSendResult match {
                 case Left(err) =>
+                  info(s"TransactionalId $transactionalId failing transition to $newMetadata due to $err")
```

Comment:
ack. the only thing is that we don't have the `epochAndMetadata` object in this context.